incorrect warning for bits unicode-data.h

`/opt/rh/gcc-toolset-14/root/usr/include/c++/14/bits/unicode-data.h:1:1: warning: no private include name for @headername mapping`

Getting this warning on each file parsed. I am using iwyu 0.23

```
[build@339322bb8ba2 build_output]$ include-what-you-use --version
include-what-you-use 0.23 (git:30e221f) based on clang version 19.1.7 (RESF 19.1.7-1.el9)
[build@339322bb8ba2 build_output]$ include-what-you-use -print-resource-dir
/usr/lib/clang/19
[build@339322bb8ba2 build_output]$ clang++ --version
clang version 19.1.7 (RESF 19.1.7-1.el9)
Target: x86_64-redhat-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Configuration file: /etc/clang/x86_64-redhat-linux-gnu-clang++.cfg
[build@339322bb8ba2 build_output]$ clang++ -print-resource-dir
/usr/bin/../lib/clang/19
```

I understand iwyu can give incorrect warning. Problem is I am unable to suppress this warning at all. I tried adding 
`{ "include": ["<bits/unicode-data.h>", "private", "<format>", "public"] }`
`{ "include": ["\"bits/unicode-data.h\"", "private", "<format>", "public"] }`
but both didn't work

--- a/iwyu_preprocessor.cc
+++ b/iwyu_preprocessor.cc
@@ -17,7 +17,6 @@
 #include "clang/AST/Decl.h"
 #include "clang/Basic/IdentifierTable.h"
 #include "clang/Basic/TokenKinds.h"
-#include "clang/Lex/HeaderSearch.h"
 #include "clang/Lex/MacroInfo.h"
 #include "clang/Lex/Token.h"
 #include "iwyu_ast_util.h"
@@ -362,12 +361,23 @@ void IwyuPreprocessorInfo::HandlePragmaComment(SourceRange comment_range) {
 }
 
 void IwyuPreprocessorInfo::ProcessHeadernameDirectivesInFile(
-    SourceLocation file_beginning) {
+    const string& quoted_private_include, SourceLocation file_beginning) {
+  if (quoted_private_include.empty()) {
+    return;
+  }
+
   SourceLocation current_loc = file_beginning;
   OptionalFileEntryRef file = GetFileEntry(current_loc);
   if (!file) {
     return;
   }
+
+  // Make sure we have a mappable name.
+  CHECK_(IsQuotedInclude(quoted_private_include))
+      << ": " << quoted_private_include;
+
+  // Quote @headername headers based on the current file's system-headerness.
+  bool is_angled = IsSystemHeader(file);
 
   while (true) {
     // Find any headername directive after a file directive. This is a Doxygen
@@ -392,20 +402,6 @@ void IwyuPreprocessorInfo::ProcessHeadernameDirectivesInFile(
     }
     after_text = after_text.substr(0, close_brace_pos);
     vector<string> public_includes = Split(after_text, ",", 0);
-
-    // Lookup the current filename as a quoted include.
-    bool is_angled = IsSystemHeader(file);
-    string include_name = preprocessor_.getHeaderSearchInfo()
-                              .getIncludeNameForHeader(*file)
-                              .str();
-    if (include_name.empty()) {
-      // Sometimes the preprocessor state can't resolve the include name; don't
-      // map empty names in that case.
-      // TODO: figure out a bullet-proof way to get include name from file.
-      Warn(file_beginning, "no private include name for @headername mapping");
-      return;
-    }
-    string quoted_private_include = AddQuotes(include_name, is_angled);
 
     // Generate mappings from the private to all public names.
     for (string& public_include : public_includes) {
@@ -851,7 +847,7 @@ void IwyuPreprocessorInfo::FileChanged_EnterFile(
   if (IsSpecialFile(new_file))
     return;
 
-  ProcessHeadernameDirectivesInFile(file_beginning);
+  ProcessHeadernameDirectivesInFile(include_name_as_written, file_beginning);
 
   // The first non-special file entered is the main file.
   if (main_file_ == nullptr)

--- a/iwyu_preprocessor.h
+++ b/iwyu_preprocessor.h
@@ -258,8 +258,9 @@ class IwyuPreprocessorInfo : public clang::PPCallbacks,
   // Determine if the comment is a pragma, and if so, process it.
   void HandlePragmaComment(clang::SourceRange comment_range);
 
-  // Process @headername directives in a file.
-  void ProcessHeadernameDirectivesInFile(clang::SourceLocation file_beginning);
+  // Process @headername directives in a file include as include_name.
+  void ProcessHeadernameDirectivesInFile(const string& include_name,
+                                         clang::SourceLocation file_beginning);
 
   // Checks whether it's OK to use the given macro defined in file defined_in.
   void ReportMacroUse(const string& name,

